src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectNative.cs

```diff
@@ -35,9 +33,9 @@ internal class TdsParserStateObjectNative : TdsParserStateObject
 
         internal override uint Status => _sessionHandle != null ? _sessionHandle.Status : TdsEnums.SNI_UNINITIALIZED;
 
-        internal override object SessionHandle => _sessionHandle;
+        internal override SessionHandle SessionHandle => SessionHandle.FromNativeHandle(_sessionHandle);
```

Comment:
This Property accessor could be moved into the Base class now since it's no longer an `object` but a common concrete type. You could take care of it in a PR after this one is merged.

Reply:
The native and managed implementations are different and i wouldn't expect it to ever be default. If it comes up while i'm working on other things i'll address it but at the moment i think it's ok.

Reply:
OK